### PR TITLE
Adjust git sparse checkout options

### PR DESCRIPTION
**Pull request by satmandu**
- This is all in service of trying to get rid of the i686 container sparse checkout error messages on opening a container.
image

Works properly:
- [x] `x86_64`
- [x] `i686`
- [x] `armv7l` 

### Run the following to get this pull request's changes locally for testing.
```
CREW_REPO=https://github.com/satmandu/chromebrew.git CREW_BRANCH=git_sparse crew update
```

### CODE REVIEW

**Comment by Zopolis4** [comment]
Does this fix the issue? Fundamentally the only cause should be that the ARCH value is wrong-- messing with the way we set it is not an ideal solution.

**Comment by satmandu** [comment]
> Does this fix the issue? Fundamentally the only cause should be that the ARCH value is wrong-- messing with the way we set it is not an ideal solution.

I can't tell yet if it fixes the issue. It only happens when we first open the container.

Do the changes seem sane?

File: bin/crew [hunk 1]
```diff
@@ -434,9 +434,14 @@ def update
   abort "'crew update' is used to update crew itself. Use 'crew upgrade <package1> [<package2> ...]' to update specific packages.".orange if @pkg_name
 
   # update package lists
+  @other_arches = %w[i686 x86_64 armv7l]
+  @other_arches.reject! { |n| n == ARCH }
   Dir.chdir(CREW_LIB_PATH) do
     # Set sparse-checkout folders.
-    system "git sparse-checkout set packages manifest/#{ARCH} lib commands bin crew tests tools"
+    system "git sparse-checkout set --no-cone \"/*\" bin commands crew lib \"/manifest/#{ARCH}/*\" packages tests tools"
```
Comment (uberhacker) [inline bin/crew:441]: `--no-clone`?

Comment (satmandu) [reply bin/crew:441]: No it's `cone` referring to the new sparse algorithm we are avoiding.

Comment (satmandu) [reply bin/crew:441]: https://stackoverflow.com/questions/9572407/git-sparse-checkout-with-exclusion#26129711

**Comment by satmandu** [comment]
Also, `ARCH` shouldn't be setup wrong:
```
chronos@cheekon-i686 /usr/local/lib/crew $ crew const | grep ARCH
ARCH=i686
ARCH_LIB=lib
CPU_SUPPORTED_ARCH=["i686", "x86_64"]
CREW_ARCH_FLAGS=
CREW_ARCH_FLAGS_OVERRIDE=
CREW_NOT_SHRINK_ARCHIVE=false
KERN_ARCH=i686
chronos@cheekon-i686 /usr/local/lib/crew $ uname -m
i686
chronos@cheekon-i686 /usr/local/lib/crew $
```

**Comment by Zopolis4** [comment]
I'm thinking the issue might be in the setup of the docker container, before setarch is available. The error message suggests that manifest/x86_64 was originally set, then later manifest/i686 was set, but manifest/x86_64 still remains. This would be consistent with the issue occurring in the initial installation.

**Comment by satmandu** [comment]
> I'm thinking the issue might be in the setup of the docker container, before setarch is available. The error message suggests that manifest/x86_64 was originally set, then later manifest/i686 was set, but manifest/x86_64 still remains. This would be consistent with the issue occurring in the initial installation.

I think you're right. This is probably an interaction with the `chromebrewstart` command, which might be running `crew update` from `~/.bashrc` before `/usr/local/bin/setarch \\\${SETARCH_ARCH} sudo -i -u chronos` is invoked.

**Comment by satmandu** [comment]
Will work on fixing the container startup scripts...

**Pull request closed without merging**